src/tree.rs

```diff
@@ -2525,4 +2536,48 @@ mod tree_tests {
         tree.remove_node(child_2_id, LiftChildren).unwrap();
         assert_eq!(1, tree.height());
     }
+
+    #[test]
+    fn test_cloned() {
```

Comment:
This test looks good to me, but I'd love to have another test (or just more in this test, either way is fine) that shows that the tree structures are identical as well if you can.

For example, I'd love to know that the indices of all of the nodes match those in the original tree and that the data at those indices is the same; I'd also like to know that each node has the correct child/parent relationships if possible.

Reply:
Again I completely forgot about this PR. Will do that today! :)

Reply:
Ha! No problem at all... The holiday season kept me pretty busy this year so I had also forgotten about it until I got this notification.

Reply:
It's done.

> For example, I'd love to know that the indices of all of the nodes match those in the original tree
Done (that's the i != j)

> and that the data at those indices is the same
Done (node_1 != node_2)

> I'd also like to know that each node has the correct child/parent relationships if possible.
Done, I only do the parents since the children would be obviously the same.

Reply:
>[...] it should be an Option. [...]
At this point I very much agree with this.

I wrestled with this quite a bit when I originally wrote the library.  I wanted there to be a way for the user to know that they're talking to the wrong tree instead of looking for a node that doesn't exist.  Now though, I'm not certain that's actually useful.